lib/groups.js

```diff
@@ -139,12 +179,54 @@ module.exports.createGroup = function (groupName, assessmentId, userId, authnUse
 };
 
 module.exports.leaveGroup = function (assessmentId, userId, authnUserId, callback) {
+  const params = [assessmentId, userId, authnUserId];
+  sqldb.call('group_leave', params, function (err) {
+    if (ERR(err, callback)) return;
+    callback();
+  });
+};
+
+module.exports.getAssessmentLevelPermissions = function (assessmentId, userId, callback) {
   const params = {
     assessment_id: assessmentId,
     user_id: userId,
-    authn_user_id: authnUserId,
   };
-  sqldb.query(sql.leave_group, params, function (err) {
+  sqldb.query(sql.get_assessment_level_permissions, params, function (err, result) {
+    if (ERR(err, callback)) return;
+    callback(result.rows[0]);
+  });
+};
+
+module.exports.updateGroupRoles = function (
+  requestBody,
+  assessmentId,
+  userId,
+  authnUserId,
+  callback
+) {
+  // Convert form data to valid input format for a SQL function
+  const roleKeys = Object.keys(requestBody).filter((key) => !key.startsWith('__'));
+  const uidToRoleIdMap = {};
+  for (const roleKey of roleKeys) {
+    const [roleId, uid] = roleKey.split('-');
+    if (uidToRoleIdMap[uid] === undefined) {
+      uidToRoleIdMap[uid] = [roleId];
+    } else {
+      uidToRoleIdMap[uid].push(roleId);
+    }
+  }
+
+  const roleAssignments = [];
+  Object.entries(uidToRoleIdMap).forEach((entry) => {
+    const roleAssignment = {
+      uid: entry[0],
+      group_role_ids: entry[1].map((id) => parseInt(id)),
+    };
+    roleAssignments.push(roleAssignment);
+  });
```

Comment:
Nit, this is a bit more concise:

```suggestion
  const roleAssignments = Object.entries(uidToRoleIdMap).map((entry) => ({
    uid: entry[0],
    group_role_ids: entry[1].map((id) => parseInt(id)),
  }));
```